py-polars/tests/unit/namespaces/test_string.py

```diff
@@ -537,6 +537,39 @@ def test_extract_all_many() -> None:
     assert df["foo"].str.extract_all(df["re"]).to_list() == [["a"], ["bc"], ["abc"]]
 
 
+def test_extract_groups() -> None:
```

Comment:
This is a nice complex test case. Could we also add a very simple test for both named groups and numeric groups?

Reply:
Yeah, good point - thanks. More testing is needed.

No groups was actually not handled correctly:
```python
df.select(pl.col("iso_code").str.extract_groups(""))

# shape: (1, 1)
# ┌───────────┐
# │ iso_code  │
# │ ---       │
# │ struct[1] │
# ╞═══════════╡
# │ {null}    │
# └───────────┘
```
I've just updated it to cast the original column to null
```python
df.select(pl.col("iso_code").str.extract_groups(""))

# shape: (2, 1)
# ┌───────────┐
# │ iso_code  │
# │ ---       │
# │ struct[1] │
# ╞═══════════╡
# │ {null}    │
# │ {null}    │
# └───────────┘

df.select(pl.col("iso_code").str.extract_groups("")).schema
# {'iso_code': Struct([Field('iso_code', Null)])}
```
Not sure what the "correct" field name should be in this case.